src/charm.py

```diff
@@ -500,52 +497,6 @@ def _on_peer_relation_departed(self, event: RelationDepartedEvent) -> None:
         # Update the sync-standby endpoint in the async replication data.
         self.async_replication.update_async_replication_data()
 
-    def _on_pgdata_storage_detaching(self, _) -> None:
```

Comment:
Step down seems to cause stuck tests on juju 2 ([db](https://github.com/canonical/postgresql-operator/actions/runs/13260012137/job/37014541847?pr=672) and [replication](https://github.com/canonical/postgresql-operator/actions/runs/13260012137/job/37014547405?pr=672)). Since now we have means to reinitialise RAFT when stuck, we should be able to remove this.